samcli/lib/package/s3_uploader.py

```diff
@@ -144,6 +145,43 @@ def upload_with_dedup(
 
         return self.upload(file_name, remote_path)
 
+    def delete_artifact(self, remote_path: str, is_key=False):
+        """
+        Deletes a given file from S3
+        :param remote_path: Path to the file that will be deleted
```

Comment:
question: the remote path should be determined by the children classes of the S3Uploader .. am I correct?

Reply:
Yes the remote_path is basically the key when we parse S3 artifacts from the template file and if only the file_name is provided then we create the key using s3_prefix + remote_path.